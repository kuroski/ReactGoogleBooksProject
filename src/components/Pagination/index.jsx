import React, {Component} from 'react'
import styled from 'styled-components'
import ReactPaginate from 'react-paginate'

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
`

const propTypes = {
  currentPage: React.PropTypes.number.isRequired,
  pageCount: React.PropTypes.number.isRequired,
  onPageChange: React.PropTypes.func.isRequired,
}

class Pagination extends Component {
  constructor(props) {
    super(props)
    this.changePage = this.changePage.bind(this)
  }

  changePage(index) {
    if(index.selected === this.props.currentPage) return
    this.props.onPageChange(index.selected)
    window.scrollTo(0, 0)
  }

  render() {
    return (
      <Container>
        <ReactPaginate
          containerClassName="c-pagination"
          pageClassName="c-pagination__page"
          pageCount={this.props.pageCount}
          marginPagesDisplayed={2}
          pageRangeDisplayed={5}
          onPageChange={this.changePage}
          activeClassName="c-pagination__page--current"
          disabledClassName="c-pagination__page--disabled"
          previousLabel=""
          nextLabel=""
        />
      </Container>
    )
  }
}

Pagination.propTypes = propTypes

export default Pagination
