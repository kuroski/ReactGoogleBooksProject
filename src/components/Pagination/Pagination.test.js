import React from 'react'
import Pagination from './'
import renderer from 'react-test-renderer'

describe('BookStore', () => {
  it('renders correctly', () => {
    const PaginationTree = renderer.create(
      <Pagination currentPage={0} pageCount={0} onPageChange={() => ''} />
    ).toJSON()
    expect(PaginationTree).toMatchSnapshot()
  })
})