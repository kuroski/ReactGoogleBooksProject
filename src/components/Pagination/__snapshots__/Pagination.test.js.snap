exports[`BookStore renders correctly 1`] = `
<div
  className="dLOtxE">
  <ul
    className="c-pagination">
    <li
      className="previous c-pagination__page--disabled">
      <a
        className={undefined}
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0">
        
      </a>
    </li>
    <li
      className="next">
      <a
        className={undefined}
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0">
        
      </a>
    </li>
  </ul>
</div>
`;
